collect_app/src/main/java/org/odk/collect/android/widgets/utilities/DateTimeWidgetUtils.java

```diff
@@ -0,0 +1,114 @@
+package org.odk.collect.android.widgets.utilities;
+
+import android.os.Build;
+import android.os.Bundle;
+
+import org.javarosa.core.model.FormIndex;
+import org.javarosa.core.model.data.TimeData;
+import org.joda.time.DateTime;
+import org.joda.time.LocalDateTime;
+import org.odk.collect.android.activities.FormEntryActivity;
+import org.odk.collect.android.application.Collect;
+import org.odk.collect.android.fragments.dialogs.BikramSambatDatePickerDialog;
+import org.odk.collect.android.fragments.dialogs.CopticDatePickerDialog;
+import org.odk.collect.android.fragments.dialogs.CustomTimePickerDialog;
+import org.odk.collect.android.fragments.dialogs.EthiopianDatePickerDialog;
+import org.odk.collect.android.fragments.dialogs.FixedDatePickerDialog;
+import org.odk.collect.android.fragments.dialogs.IslamicDatePickerDialog;
+import org.odk.collect.android.fragments.dialogs.MyanmarDatePickerDialog;
+import org.odk.collect.android.fragments.dialogs.PersianDatePickerDialog;
+import org.odk.collect.android.javarosawrapper.FormController;
+import org.odk.collect.android.logic.DatePickerDetails;
+import org.odk.collect.android.utilities.DialogUtils;
+import org.odk.collect.android.utilities.ThemeUtils;
+
+public class DateTimeWidgetUtils {
```

Comment:
What is the reason to have DateTimeWidgetUtils and DateTimeUtils as separate classes? Shouldn't it be just one class?

Reply:
I just thought that the methods in `DateTimeUtils` are very much specific to getting a particular format of Date and Time Data. Although the names of both classes are very similar.

Reply:
Should we have a single class here, just `DateTimeUtils`?

Reply:
I think we should review those classes because even if it make sense to keep them both at least we need to move some methods. For example in DateTimeWidgetUtils you have getCurrentDate()/getTimeData() where it would be better to have it in DateTimeUtils on the other hand in DateTimeUtils you have getDateTimeLabel() where it should probably be in DateTimeWidgetUtils because it's a label for widgets.

Reply:
Okay sure! I will review them.

Reply:
Is this cleaning done? Because I doubt.

Reply:
Yes done in [e2765b1](https://github.com/getodk/collect/pull/3975/commits/e27c5b1f7b511e2919db72a246e3f398320f6f49)

Reply:
What about: getDatePickerDetails/getDateTimeLabel/getGregorianDateTimeLabel shouldn't it also be a part of DateTimeWidgetUtils?

Reply:
Right @grzesiek2010, they should be a part of `WidgetUtils`